test/integration/associations/belongs-to-many.test.js

```diff
@@ -616,6 +616,93 @@ describe(Support.getTestDialectTeaser('BelongsToMany'), function() {
         expect(tasks[0].title).to.equal('wat');
       });
     });
+
+    it('using scope to set associations', function() {
+      var ItemTag = this.sequelize.define('ItemTag', {
+                      id : {
+                            type: DataTypes.INTEGER,
+                            primaryKey: true,
+                            autoIncrement: true
+                           },
+                      tag_id: {
+                            type: DataTypes.INTEGER,
+                            unique: false
+                      },
+                      taggable: {
+                        type: DataTypes.STRING,
+                      },
+                      taggable_id: {
+                        type: DataTypes.INTEGER,
+                        unique: false
+                      }
+      }),
+      Tag = this.sequelize.define('Tag', {
+        id : {
+              type: DataTypes.INTEGER,
+              primaryKey: true,
+              autoIncrement: true
+             },
+        name: DataTypes.STRING
+      }),
+      Comment = this.sequelize.define('Comment', {
+        id : {
+              type: DataTypes.INTEGER,
+              primaryKey: true,
+              autoIncrement: true
+             },
+        name: DataTypes.STRING
+      }),
+      Post = this.sequelize.define('Post', {
+        id : {
+              type: DataTypes.INTEGER,
+              primaryKey: true,
+              autoIncrement: true
+             },
+        name: DataTypes.STRING
+      });
+
+      Post.belongsToMany(Tag, {
+        through: {
+          model: ItemTag,
+          unique: false,
+          scope: {
+            taggable: 'post'
+          }
+        },
+        foreignKey: 'taggable_id'
+      });
+
+      Comment.belongsToMany(Tag, {
+        through: {
+          model: ItemTag,
+          unique: false,
+          scope: {
+            taggable: 'comment'
+          }
+        },
+       foreignKey: 'taggable_id'
+      });
+
+      return this.sequelize.sync({ force: true }).then(function() {
+        return Promise.all([
+          Post.create({ name: 'post1' }),
+          Comment.create({ name: 'comment1' }),
+        ]);
+      }).bind({}).spread(function(post, comment) {
+        this.post = post;
+        this.comment = comment;
+        return Tag.create({ name: 'tag1' });
+      }).then(function(tag) {
+        this.tag = tag;
+        return this.post.setTags([tag]);
```

Comment:
If I do it like that, I can't reproduce the issue, I need to add the tag to the post and then, add it to the comment.